### PR TITLE
Fix list today bugs

### LINKED ISSUE #163: [PE-D][Tester C] Adding medicine repetition out of bounds
According to the userguide, the repetition can only be a number from 1-7 and hence should show a index out of bounds exception. However, when inputting repetition 8, the medicine is still added.

-------------
Labels: `severity.Low` `type.FeatureFlaw`
original: cayennegoh/ped#3

### LINKED ISSUE #172: [PE-D][Tester A] Program crash when incorrect command
If the user accidentally keys in -m -a in the command to list the medication taken today, the program crashes. It would be great if the program could throw an exception for this type of command error.

-------------
Labels: `type.FunctionalityBug` `severity.High`
original: Haotian199/ped#3

### LINKED ISSUE #176: [PE-D][Tester B] 'list -t all' command ignores ignores subsequent flags
_'list -t all'_ command ignores ignores subsequent flags i.e., entering _"list -t all -m"_ does not display all medications to be taken only in morning - instead it just shows all medications like _'list -t all'_ alone

Photo showing all flags (_-m -a and & -e_) being ignored by _'list -t all'_ 

If this functionality is not desired, I would recommend explicitly specifying in user guide that these flags should not be added after _'list -t all'_  and also displaying an error message in the command line interface if this attempted, instead of listing drugs.

List help message makes it seem to a user that they should be able to enter _'list -t all -flag'_  

I have listed this bug as medium severity as a user may genuinely desire to see all drugs to be taken at a specific time regardless of day (they can see drugs to be taken at a specific time, but only for one day)

-------------
Labels: `type.FunctionalityBug` `severity.Medium`
original: LWachtel1/ped#2

### LINKED ISSUE #199: [PE-D][Tester D] Exit command missing
Include `exit` command in UG.

-------------
Labels: `severity.High` `type.DocumentationBug`
original: fxe025/ped#10

### PR DESCRIPTION
**Pull request by T0nyLin**
# Checklist for feature implementation and modification
Fixes #163, fixes #172, fixes #176, fixes #199
- [X] Added relevant milestone (under milestone tab on the right)
- Implementation
  - [ ] Sources (if any) credited under both relevant section(s) and Developer Guide (Acknowledgement)
  - [ ] JUnit Test(s)
- Documentation
  - [X] JavaDoc for classes and methods
  - Developer Guide
    - [ ] Design (Can add as a todo statement for later on)
    - [ ] Implementation (Can add as a todo statement for later on)
    - [ ] Relevant User Story present
    - [ ] Non-functional requirement(s) that the feature addresses (if any)
    - [ ] Glossary of the (technical) term(s) used (if any)
    - [ ] Specialised instructions for manual testing (if any)
  - User Guide
    - [X] Description and usage of feature (if any)
  - Contributions (under `/docs/team/*.md`)
    - [X] Code/Feature/Enhancement contribution
    - [X] Developer Guide contribution
    - [X] User Guide contribution
    - [ ] Misc and team-based admin tasks (if any)

# Checklist for non-implementation related PR
Fixes #163, fixes #172, fixes #176, fixes #199
- [ ] Added relevant milestone (under milestone tab on the right)
- Documentation (Contributions)
  - [ ] Developer Guide contributions (if any)
  - [ ] User Guide contributions (if any)
  - [ ] Review contributions (if any)
  - [X] Miscellaneous and team-based admin tasks (if any)

# Additional Information

Nil

### CODE REVIEW

File: src/main/java/meditracker/command/ListCommand.java [hunk 1]
```diff
@@ -66,38 +65,67 @@ public void execute() {
...
-                DailyMedicationManager.printTodayMedications(MedicationManager.getMedications(),
-                        eveningMedications, "Evening:");
+
+        // checks if user added extra flags or words after "list -t all"
+        if (listTypeString.equals("all") && parsedArguments.size() > 1) {
+            Ui.showErrorMessage(String.format("List type -> \"%s\" not compatible with " +
+                    "\"list -t all\" command.", period));
+        } else {
+            switch (listTypeString) {
+            case "all":
```
Comment (wenenhoe) [inline src/main/java/meditracker/command/ListCommand.java:75]: Might be better to move L70 condition check into this switch case so the switch statement does not have to be nested an additional layer

```suggestion
            case "all":
                if (parsedArguments.size() > 1) {
                    Ui.showErrorMessage(String.format("List type -> \"%s\" not compatible with " +
                    "\"list -t all\" command.", period));
                    return;
                }
```

**Review by wenenhoe — CHANGES_REQUESTED** [review]
Thanks for fixing the bugs from PE-D! Just some minor changes for your consideration

File: src/main/java/meditracker/command/ListCommand.java (hunk 1 above)
Comment (T0nyLin) [reply src/main/java/meditracker/command/ListCommand.java]: Thank you for the suggestion, implemented in 98ba095

File: src/main/java/meditracker/ui/Ui.java [hunk 2]
```diff
@@ -156,16 +156,22 @@ public static void printSpecificMed(Medication medication) {
...
+                if (period.equals("Morning:")) {
+                    intakeDose = medsList.get(index).getDosageMorning();
+                } else if (period.equals("Afternoon:")) {
+                    intakeDose = medsList.get(index).getDosageAfternoon();
+                } else {
+                    intakeDose = medsList.get(index).getDosageEvening();
+                }
+            } catch (IndexOutOfBoundsException e) {
+                return;
             }
```
[the remark this answers is not in the document]
Comment (T0nyLin) [reply src/main/java/meditracker/ui/Ui.java]: Thank you for the suggestion, implemented in 98ba095

**Pull request merged**